libs/common/CMakeLists.txt

```diff
@@ -1,18 +1,21 @@
-add_library(libs_common
-    files.cpp
-    )
-
-target_link_libraries(libs_common
-    logger
-    boost
-    )
+# Copyright Soramitsu Co., Ltd. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
 
 add_library(common INTERFACE)
-
 target_link_libraries(common INTERFACE
-    boost
-    )
-target_include_directories(common INTERFACE
-    ${PROJECT_SOURCE_DIR}/libs/common
-    )
+  boost
+  )
+
+add_library(libs_files
+  files.cpp
+  )
+target_link_libraries(libs_files INTERFACE
+  logger
+  common
+  )
 
+add_library(libs_timeout INTERFACE)
+target_link_libraries(libs_timeout INTERFACE
+  common
+  rxcpp
+  )
```

Comment:
Also, pls, add library for datetime with a separation to cpp and header

Reply:
There're two functions there and one of it is templated, the second one is simple enough for being inlined. Moreover return type is deduced here. 
Thus there's no need for such action